hail/python/hail/matrixtable.py

```diff
@@ -4050,12 +4050,16 @@ def make_table(self, separator='.') -> Table:
         if not (len(self.col_key) == 1 and self.col_key[0].dtype == hl.tstr):
             raise ValueError("column key must be a single field of type str")
 
-        col_key_field = list(self.col_key)[0]
-        col_keys = [k[col_key_field] for k in self.col_key.collect()]
+        col_keys =  self.col_key[0].collect()
 
         duplicates = [k for k, count in Counter(col_keys).items() if count > 1]
         if duplicates:
             raise ValueError(f"column keys must be unique, found duplicates: {', '.join(duplicates)}")
+        for c in col_keys:
```

Comment:
You could avoid the second scan by reusing the counter:
```
counts = Counter(col_keys)
duplicates = [k for k, count in counts.items() if count > 1]
if duplicates:
  ...
if counts[None] > 0:
  ...
```
But if there is more than one missing key, won't this throw the duplicate key error?

Reply:
good call, fixed.